### ISSUE TITLE
LOSS

### ISSUE THREAD

**Issue by yishunzhijian** [miemie2013/Keras-DIOU-YOLOv3#4]
用模式0从头训练20epochs得到ep000020-loss42.469-val_loss42.163.h5文件，然后运行1_lambda2model.py转换成aaaa_bgr.h5文件，用aaaa_bgr.h5训练模式1。
请问损失直接出现nan这是怎么回事？yolo_bgr_mAP_46.h5这个权重文件是采用什么训练方式得到的？

**Comment by miemie2013** [comment]
尝试一下减小学习率。转换来的不是训练来的。

**Comment by yishunzhijian** [comment]
> 尝试一下减小学习率。转换来的不是训练来的。
已经减到0.0000001还是nan..

**Comment by miemie2013** [comment]
> > 尝试一下减小学习率。转换来的不是训练来的。
> > 已经减到0.0000001还是nan..

pos_loss = respond_bbox * (0 - K.log(pred_conf))
neg_loss = respond_bgd  * (0 - K.log(1 - pred_conf))
改成
pos_loss = respond_bbox * (0 - K.log(pred_conf + 1e-9))
neg_loss = respond_bgd  * (0 - K.log(1 - pred_conf + 1e-9))

**Comment by yishunzhijian** [comment]
可以训练了，谢谢。一个建议，可以加个中断后能继续训练的代码，因为学习率得自己调。
yolo_bgr_mAP_46.h5这个权重文件是通过1_lambda2model.py转换过来的吗？

**Comment by miemie2013** [comment]
> 可以训练了，谢谢。一个建议，可以加个中断后能继续训练的代码，因为学习率得自己调。
> yolo_bgr_mAP_46.h5这个权重文件是通过1_lambda2model.py转换过来的吗？

因为你不能预先知道训练几个epoch之后loss不再下降，所以手动调其实挺好的。从其它仓库搞到的。

**Comment by yishunzhijian** [comment]
有学习率衰减，loss不降就自动调低学习率。（中断后可以继续训练）
